msal/application.py

```diff
@@ -176,6 +176,10 @@ class ClientApplication(object):
     REMOVE_ACCOUNT_ID = "903"
 
     ATTEMPT_REGION_DISCOVERY = True  # "TryAutoDetect"
+    _TOKEN_SOURCE = "token_source"
+    _TOKEN_SOURCE_IDP = "idp"
```

Comment:
nit: IDP might be an acronym that is too specific to Identity geeks. Consider using "IdentityProvider" string in constants that are part of the public api.

Reply:
I thought you mentioned the term `IdP` in the feature request #602. But yeah, how about this?

```suggestion
    _TOKEN_SOURCE_IDP = "identity_provider"
```

W.R.T. the leading underscore naming, indeed it means private in Python convention. My thought here is that since MSAL Python returns a plain-old dictionary `{"access_token": "...", ..., "token_source": "identity_provider"}` (rather than returning an AuthenticationResult class), those visible string literals themselves (such as `token_source` and `identity_provider`) become the *real* public API. Therefore, the `_TOKEN_SOURCE` and `_TOKEN_SOURCE_IDP` names are mainly just MSAL Python internal constants which we use to avoid duplicating magic strings everywhere.

Bottom line is, I think we can go with the _PRIVATE notation for now. If a need arises in the future that we would want to promote them to become public, we can always do that later.